pkg/render/apiserver.go

```diff
@@ -684,11 +688,55 @@ func (c *apiServerComponent) secretsClusterRole() []client.Object {
 		&rbacv1.ClusterRoleBinding{
 			TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
 			ObjectMeta: metav1.ObjectMeta{
-				Name: name,
+				Name: APIServerSecretsRBACName,
 			},
 			RoleRef: rbacv1.RoleRef{
 				Kind:     "ClusterRole",
-				Name:     name,
+				Name:     APIServerSecretsRBACName,
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      APIServerServiceAccountName(c.cfg.Installation.Variant),
+					Namespace: rmeta.APIServerNamespace(c.cfg.Installation.Variant),
+				},
+			},
+		},
+	}
+}
+
+// secretsRole provides the tigera API server with the ability to read secrets from the API server's namespace.
+func (c *apiServerComponent) secretsRole() []client.Object {
```

Comment:
I think this code would be more clear if we named this `secretsRBAC` and the other one `multiTenantSecretsRBAC` - that way it's clear the distinction between the two.